react/features/gifs/reducer.js

```diff
@@ -0,0 +1,66 @@
+
+import { ReducerRegistry } from '../base/redux';
+
+import {
+    ADD_GIF_FOR_PARTICIPANT,
+    HIDE_GIF_FOR_PARTICIPANT,
+    REMOVE_GIF_FOR_PARTICIPANT,
+    SET_GIF_DRAWER_VISIBILITY
+} from './actionTypes';
+
+const initialState = {
+    gifList: new Map(),
+    drawerVisible: false
+};
+
+ReducerRegistry.register(
+    'features/gifs',
+    (state = initialState, action) => {
+        switch (action.type) {
+        case ADD_GIF_FOR_PARTICIPANT: {
+            const newList = state.gifList;
+
+            newList.set(action.participantId, {
```

Comment:
Did you make sure the mapping is cleaned when a user leaves the meeting even when there are pending gifs?

Reply:
Although, I'm not sure it's a problem. So if we add gif for a participant that's already left, it's just gonna add it to the store, but it won't cause any re-renders. And the timer will be set, so it will get removed. And if a participant leaves and it has a gif the gif will be removed after the timer, so it will not remain there

Reply:
Ack, the timer will clean it up, so we're good.